plugins/system/sef/src/Extension/Sef.php

```diff
@@ -188,6 +245,51 @@ function ($match) use ($base, $protocols) {
         $this->getApplication()->setBody($buffer);
     }
 
+    /**
+     * Remove unnecessary index.php from URLs built in Joomla
+     *
+     * @param   Router  &$router  Router object.
+     * @param   Uri     &$uri     Uri object.
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function removeIndexphp(&$router, &$uri)
+    {
+        $path = $uri->getPath();
+
+        if (substr($path, -9) == 'index.php') {
+            $uri->setPath(substr($path, 0, -9));
+        }
+    }
+
+    /**
+     * Redirect to a URL without unnecessary index.php
+     *
+     * @param   Router  &$router  Router object.
+     * @param   Uri     &$uri     Uri object.
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function enforceRemovedIndexphp(&$router, &$uri)
+    {
+        // We only want to redirect on GET requests
+        if ($_SERVER['REQUEST_METHOD'] != 'GET') {
```

Comment:
```suggestion
        if ($this->app->getInput()->getMethod() != 'GET') {
```

If i remember right this is the same thing but properly using the Input abstraction

Reply:
I'm going to trust you on this one.